### PR TITLE
Regex change for domainName field

### PR DESCRIPTION
**Pull request by saikiranakula-amzn**
*Description of changes:*
Regex validation check if the domainName is up to DNS naming standards
Testing:
int main() {
    std::cout << "test regex" << "\n";

    //std::cout << "a.com" << "\n";
    std::cout << isValidDomain("a.com") << "\n";
    //std::cout << isValidDomain("ab.com") << "\n";
    std::cout << isValidDomain("ab.toto-abc.com") << "\n"; 
    std::cout << isValidDomain("p/") << "\n";
    std::cout << isValidDomain("test4.gmsa-pentest.com") << "\n";
    std::cout << isValidDomain("-geeksforgeeks.org") << "\n";
    std::cout << isValidDomain("contoso.com") << "\n";
    std::cout << isValidDomain(".org") << "\n";
    
    return 0;
}
~                                                                                                                                                               
"regex.cpp" 49L, 1331B written
[ec2-user@EC2AMAZ-Ll40tS ~]$ g++ -o test regex.cpp 
[ec2-user@EC2AMAZ-Ll40tS ~]$ ./test 
test regex
test1
test1
fail0
test1
fail0
test1
fail0

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jamolina — APPROVED** [review]
Approved

File: api/tests/gmsa_test_client.cpp [hunk 1]
```diff
@@ -581,7 +581,19 @@ int parse_credspec_domainless_test(std::string credspec)
...
+    }
+    return  false;
+}
+
+int validate_domain()
+{
+    return (isValidDomain("a.com") && isValidDomain("ab.toto-abc.com") &&
+             !isValidDomain("p/") && isValidDomain("test4.gmsa-pentest.com") &&
+             !isValidDomain ("-testdomain.org") &&  isValidDomain("contoso.com") &&
+             !isValidDomain(".org"));
```
Comment (as14692) [inline api/tests/gmsa_test_client.cpp:596]: nit: I'd recommend clubbing all valid and invalid cases separately for readability 
```
string validDomains[] = {
    "a.com",
    "ab.toto-abc.com", 
    "test4.gmsa-pentest.com",
    "contoso.com"
  };

string invalidDomains[] = {  
    "p/",
    "-testdomain.org",
    ".org"
  };
``` 
Assert `validDomains` needs to be valid and `invalidDomains` needs to be invalid when `isValidDomain` is called

File: auth/kerberos/src/krb.cpp [hunk 2]
```diff
@@ -1126,13 +1126,20 @@ std::string retrieve_secret_from_ecs_config(std::string ecs_variable_name)
...
         std::string key = results[0];
         std::string value = results[1];
         if ( ecs_variable_name.compare( key ) == 0 )
         {
             value.erase( std::remove( value.begin(), value.end(), '"' ), value.end() );
+
+            if( contains_invalid_characters_in_ad_account_name(value))
+            {
+                std::cout << getCurrentTime() << '\t' << "invalid domain controller name" <<
+                    std::endl;
```
Comment (as14692) [inline auth/kerberos/src/krb.cpp:1139]: nit: Consider adding `value` to logs for better debugging (if `value` is safe to expose in log files)

File: api/tests/gmsa_test_client.cpp (hunk 1 above)
Comment (saikiranakula-amzn) [reply api/tests/gmsa_test_client.cpp:596]: we need to redo all the unit tests with a framework we can take an AI for then

File: auth/kerberos/src/krb.cpp (hunk 2 above)
Comment (saikiranakula-amzn) [reply auth/kerberos/src/krb.cpp:1139]: credpec is a large json object, we shouldn't do it unless it is necessary

Comment (as14692) [reply auth/kerberos/src/krb.cpp:1139]: afaik the variable `value` in this context is the ad account name and not the whole credspec object

**Pull request merged**